### PR TITLE
Add Claude skills for corrrectness validation

### PR DESCRIPTION
**Pull request by haok1402**
Quick regression check for code changes — run 15 training steps on both the base and feature branch, compare loss curves, and flag if anything diverged.

## What's in here

- A Claude skill (`.claude/skills/correctness-validation/`) that walks the agent through the full workflow: figure out which models are affected, set up data, run both branches via git worktree, and compare the logs.
- Per-model setup and validation scripts for DeepSeek-V2-Lite (4 GPU) and Qwen3-30B-A3B (16 GPU), plus SLURM-aware launch scripts.
- `compare.py` parses stdout logs and checks cross-entropy loss, load-balance loss, and gradient norm step-by-step against a configurable tolerance (default 1e-3).
- `save_interval` and `save_location` in `TrainingCfg` are now optional (default `None`). Setting `save_location` without `save_interval` lets you load a pretrained checkpoint without ever writing one back — exactly what the validation scripts need to run a quick 15-step check without cluttering the checkpoint directory.

### CODE REVIEW

File: pithtrain/tasks/pretrain_language_model.py [hunk 1]
```diff
@@ -284,6 +284,7 @@ def save_checkpoint(cfg: PretrainLanguageModelCfg, ctx: PretrainLanguageModelCtx
     DTensors are kept as CPU DTensors and DCP saves each rank's shard.
     """
     stdout = ctx.logging.stdout
+    assert cfg.training.save_location is not None
```
[the remark this answers is not in the document]
Comment (haok1402) [reply pithtrain/tasks/pretrain_language_model.py:287]: we'll do a fix later to better handle the exception for all the assert inside this codebase.

**Review by MasterJH5574 — APPROVED** [review]
LGTM, good to have the first skill 🎉
